app/legacy/dao/api_keys_dao.py

```diff
@@ -69,6 +65,26 @@ async def get_api_keys(service_id: UUID4) -> Sequence[Row[Any]]:
             raise NonRetryableError('Uexpected SQLAlchemy error during service API keys lookup.') from e
 
 
+@db_retry
+@alru_cache(maxsize=1024, ttl=FIVE_MINUTES)
+async def _get_api_keys_for_service(service_id: UUID4) -> Sequence[Row[Any]]:
+    """Retryable and cached function to get a ApiKey row.
+
+    Args:
+        service_id (UUID4): The api key UUID
+
+    Returns:
+        Row[Any]: A ApiKey row
+    """
+    legacy_api_keys = metadata_legacy.tables['api_keys']
+    stmt = select(legacy_api_keys).where(legacy_api_keys.c.service_id == service_id)
+
+    async with get_read_session_with_context() as session:
+        result = await session.execute(stmt)
+
+    return result.fetchall()
```

Comment:
Prematurely approved. This needs to be fixed first. Once this is fixed, it should be good to go.